### PR TITLE
test: add unit tests and fix null handling for RepositoryMapperImpl #43

### PR DESCRIPTION
**Pull request by PedroAlex65**
Hi @Regyl! I've completed the task for issue #43.

Key changes:

Bug Fix: Added a guard clause in RepositoryMapperImpl to handle null inputs safely.

New Tests: Implemented RepositoryMapperImplTest to validate both null and full object mapping scenarios using Mockito for date consistency.

All tests are passing. Ready for review!"

### CODE REVIEW

File: src/test/java/com/github/regyl/gfi/mapper/RepositoryMapperImplTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.OffsetDateTime;
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
```
Comment (Regyl) [inline src/test/java/com/github/regyl/gfi/mapper/RepositoryMapperImplTest.java]: We are using `assertJ` instead of `JUnit ` here. Please follow project rules and switch to assertj
